core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -611,11 +611,33 @@ object TestUtils extends Logging {
     */
   def createOffsetsTopic(zkClient: KafkaZkClient, servers: Seq[KafkaBroker]): Unit = {
     val server = servers.head
-    createTopic(zkClient, Topic.GROUP_METADATA_TOPIC_NAME,
-      server.config.getInt(KafkaConfig.OffsetsTopicPartitionsProp),
-      server.config.getShort(KafkaConfig.OffsetsTopicReplicationFactorProp).toInt,
-      servers,
-      server.groupCoordinator.groupMetadataTopicConfigs)
+    val numPartitions = server.config.offsetsTopicPartitions
+    val replicationFactor = server.config.offsetsTopicReplicationFactor.toInt
+
+    try {
+      createTopic(
+        zkClient,
+        Topic.GROUP_METADATA_TOPIC_NAME,
+        numPartitions,
+        replicationFactor,
+        servers,
+        server.groupCoordinator.groupMetadataTopicConfigs
+      )
+    } catch {
+      case ex: TopicExistsException =>
+        val allPartitionsMetadata = waitForAllPartitionsMetadata(
+          servers,
+          Topic.GROUP_METADATA_TOPIC_NAME,
+          numPartitions
+        )
+
+        // If the topic already exists, we ensure that it has the required
+        // number of partitions and replication factor. If it has not, the
+        // exception is thrown further.
+        if (allPartitionsMetadata.size != numPartitions || allPartitionsMetadata.head._2.replicas.size != replicationFactor) {
```

Comment:
Is this sufficient validation? The topics might be leaked artifacts from other tests / retries and might contain different data than what is expected. So, we should probably validate that all partitions are empty as well.

As an alternative solution, can we delete the topic if it already exists here and re-create it?

Reply:
We do the same when we create it with the admin client in KRaft mode and I has been enough until today. The cluster is usually recreated for all tests so it can't be leaked from a previous one. This error only happens when the first creation fails for instance due to an network exception.

Reply:
I disagree because we only use this method when we create a new cluster. I could be brittle if we were using it differently. I think that you could always argue that you just want this method to ensure that the topic is present regardless of whether it has already been used or not. I will extend the javadoc to make this clear.

Reply:
Done.